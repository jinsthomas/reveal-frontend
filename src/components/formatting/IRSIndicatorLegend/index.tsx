import React from 'react';
import { Table } from 'reactstrap';
import {
  IndicatorThresholdItem,
  IndicatorThresholds,
  indicatorThresholdsIRS,
} from '../../../configs/settings';
import { percentage } from '../../../helpers/utils';

interface Props {
  indicatorThresholds: IndicatorThresholds;
}

const defaultProps: Props = {
  indicatorThresholds: indicatorThresholdsIRS,
};

const IRSIndicatorLegend = (props: Props) => {
  const { indicatorThresholds } = props;

  return (
    <div>
      <h5 className="mt-4 mb-3">IRS Conditional Formatting</h5>
      <div className="row">
        <div className="col-3">
          <Table className="text-center">
            <tbody>
              {Object.values(indicatorThresholds)
                .sort((a: IndicatorThresholdItem, b: IndicatorThresholdItem) =>
                  a.value > b.value ? 1 : -1
                )
                .map((item, index) => {
                  return (
                    <tr key={index} style={{ background: item.color }}>
                      <td>{item.name}</td>
                      <td>>=</td>
                      <td>{percentage(item.value)}</td>
                    </tr>
                  );
                })}
            </tbody>
          </Table>
        </div>
      </div>
    </div>
  );
};

IRSIndicatorLegend.defaultProps = defaultProps;

export default IRSIndicatorLegend;
