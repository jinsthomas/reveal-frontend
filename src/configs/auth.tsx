import { Providers } from '@onaio/gatekeeper';

import { DOMAIN_NAME, ONADATA_CLIENT_ID } from '../constants';

export const providers: Providers = {
  Ona: {
    accessTokenUri: 'https://stage-api.ona.io/o/token/',
    authorizationUri: 'https://stage-api.ona.io/o/authorize/',
    clientId: ONADATA_CLIENT_ID,
    redirectUri: `${DOMAIN_NAME}/oauth/callback/Ona/`,
    scopes: ['read', 'write'],
    state: 'abc',
    userUri: 'https://stage-api.ona.io/api/v1/user.json',
  },
};
