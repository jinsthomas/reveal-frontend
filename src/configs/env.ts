/** This module handles settings taken from environment variables */

/** The website name */
export const WEBSITE_NAME = process.env.REACT_APP_WEBSITE_NAME || 'Reveal';
export type WEBSITE_NAME = typeof WEBSITE_NAME;

/** The domain name */
export const DOMAIN_NAME = process.env.REACT_APP_DOMAIN_NAME || 'http://localhost:3000';
export type DOMAIN_NAME = typeof DOMAIN_NAME;

/** Do you want to enable the IRS features? */
export const ENABLE_IRS = process.env.REACT_APP_ENABLE_IRS === 'true';
export type ENABLE_IRS = typeof ENABLE_IRS;

/** Do you want to enable the Focus Investigation features? */
export const ENABLE_FI = process.env.REACT_APP_ENABLE_FI === 'true';
export type ENABLE_FI = typeof ENABLE_FI;

/** Do you want to disable login protection? */
export const DISABLE_LOGIN_PROTECTION = process.env.REACT_APP_DISABLE_LOGIN_PROTECTION === 'true';
export type DISABLE_LOGIN_PROTECTION = typeof DISABLE_LOGIN_PROTECTION;

/** The Superset API base */
export const SUPERSET_API_BASE = process.env.REACT_APP_SUPERSET_API_BASE;
export type SUPERSET_API_BASE = typeof SUPERSET_API_BASE;

/** The Superset API endpoint */
export const SUPERSET_API_ENDPOINT = process.env.REACT_APP_SUPERSET_API_ENDPOINT;
export type SUPERSET_API_ENDPOINT = typeof SUPERSET_API_ENDPOINT;

/** OPENSRP Client ID, for oAuth2 */
export const OPENSRP_CLIENT_ID = process.env.REACT_APP_OPENSRP_CLIENT_ID;
export type OPENSRP_CLIENT_ID = typeof OPENSRP_CLIENT_ID;

/** Onadata Client ID, for oAuth2 */
export const ONADATA_CLIENT_ID = process.env.REACT_APP_ONADATA_CLIENT_ID;
export type ONADATA_CLIENT_ID = typeof ONADATA_CLIENT_ID;
