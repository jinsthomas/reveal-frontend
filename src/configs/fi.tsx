/** Focus investigation configs */

/** Interface for a Focus Investigation Classification */
export interface Classification {
  code: string;
  name: string;
  description: string;
}

/** Focus Investigation case classifications */
export const FIClassifications: Classification[] = [
  {
    code: 'A1',
    description: 'Indigenous case recorded within the last year.',
    name: 'Active',
  },
  {
    code: 'A2',
    description: 'No indigenous case during the last year, but withing the last 3 years.',
    name: 'Residual Non-Active',
  },
  {
    code: 'B1',
    description: 'Receptive area but no indigenous cases within the last 3 years.',
    name: 'Cleared Receptive',
  },
  {
    code: 'B2',
    description: 'Non-receptive area.',
    name: 'Cleared Non-Receptive',
  },
];
