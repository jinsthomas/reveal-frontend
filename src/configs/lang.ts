// todo - init i18n here (or migrate everything below) and asynchronously export translations
// also - async export/imports article: https://medium.com/@WebReflection/javascript-dynamic-import-export-b0e8775a59d4

// Display Strings
export const WELCOME_TO_REVEAL = 'Welcome to Reveal';
export const GET_STARTED_MESSAGE = 'Get started by selecting an intervention below';
export const PRACTITIONER_CREATED_SUCCESSFULLY = 'Practitioner created Successfully.';
export const PRACTITIONER_EDITED_SUCCESSFULLY = 'Practitioner edited successfully.';
export const REMOVING_PRACTITIONER_FAILED = 'Removing Practitioner Failed';
export const ON_REROUTE_WITH_UNSAVED_CHANGES =
  'Unsaved Changes: please Save or Discard changes made';

export const HOME_TITLE = `Home page`;
export const IRS_TITLE = 'IRS';
export const IRS_REPORTING_TITLE = 'IRS Reporting';
export const CONDITIONAL_FORMATTING_RULES = 'Conditional formatting rules';
export const CONDITIONAL_FORMATTING_TITLE = 'Conditional Formatting';
export const FI_RESPONSE_ADHERENCE_TITLE = 'FI Response Adherence';
export const FOCUS_INVESTIGATION_START_TITLE = 'Focus Investigation Start';
export const DATE_CREATED = 'Date Created';

export const NEW_PLAN = 'New Plan';
export const ADD_PLAN = 'Add Plan';
export const CREATE_NEW_PLAN = 'Create New Plan';
export const UPDATE_PLAN = 'Update Plan';
export const CREATE_NEW_IRS_PLAN = 'Create New IRS Plan';
export const ADD_FOCUS_INVESTIGATION = 'Add Focus Investigation';

export const START_DATE = 'Start Date';
export const END_DATE = 'End Date';
export const NAME = 'Name';
export const LEGEND_LABEL = 'Legend';
export const DISCARD_CHANGES = 'Discard changes';
export const SAVE_CHANGES = 'Save changes';
export const ASSIGN_PLANS = 'Assign Plans';
export const ASSIGN_PRACTITIONERS = 'Assign Practitioners';

export const USERS = 'Users';
export const ABOUT = 'About';
export const LOGIN = 'Login';
export const SIGN_OUT = 'Sign Out';

export const MAP_LOAD_ERROR = 'Could not load the map';
export const AN_ERROR_OCCURED = 'An Error Ocurred';
export const PLEASE_FIX_THESE_ERRORS = 'Please fix these errors';
export const NO_INVESTIGATIONS_FOUND = 'No Investigations Found';
export const NO_PRACTITIONERS_ADDED_YET = 'No Practitioners Added yet';
export const PLAN_SELECT_PLACEHOLDER = 'Other plans in this area';
export const PLAN_STATUS_UPDATE_ERROR =
  'Sorry, something went wrong when we tried to update the plan status';
export const NO_PLAN_FOUND_ERROR = 'Sorry, no plan found in the cloud!';
export const SAVE_PLAN_DEFINITION_ERROR =
  'Uh oh, looks like something is (syntactically) wrong with the Plan schema';
export const SAVE_PLAN_NO_JURISDICTIONS_ERROR = 'Oops, no jurisdictions selected!';

export const SAVE_TEAM = 'Save Team';
export const SAVE_PLAN = 'Save Plan';
export const SAVE_AS_DRAFT = 'Save as a Draft';
export const SAVE_FINALIZED_PLAN = 'Save Finalized Plan';
export const SAVE_ASSIGNMENTS = 'Save Assignments';
export const NO_PLANS_LOADED_MESSAGE = 'No plans found...';
export const SELECT = 'Select';
export const SELECT_USERNAME = 'Select username';

export const PRACTITIONER = 'Practitioner';
export const PRACTITIONERS = 'Practitioners';
export const SAVE = 'Save';
export const CLEAR = 'Clear';
export const ACTIVE = 'Active';
export const ASSIGN = 'Assign';
export const NO_TEAMS_LOADED_MESSAGE = 'No teams loaded...';
export const SELECT_TEAMS_TO_ASSIGN = 'Select Teams to Assign';
export const ASSIGN_TEAMS = 'Assign Teams';
export const TEAMS_ASSIGNED = 'Teams Assigned';
export const YES = 'Yes';
export const NO = 'No';
export const EDIT = 'Edit';
export const ADD = 'Add';
export const REMOVE = 'Remove';

export const IRS_PLANS = 'IRS Plans';
export const COUNTRY = 'Country';
export const NEW_TITLE = 'New';
export const PLAN_TITLE = 'Plan';
export const PLAN_STATUS = 'Plan Status';
export const TEAMS = 'Teams';
export const TEAM = 'Team';
export const EDIT_TEAM = 'Edit Team';
export const NEW_TEAM = 'New Team';
export const TEAMS_ASSIGNMENT = 'Team Assignment';
export const SEARCH = 'Search';
export const ACTION = 'Action';
export const ACTIONS = 'Actions';
export const IDENTIFIER = 'Identifier';
export const USERNAME = 'Username';
export const VIEW = 'view';

export const REACTIVE = 'Reactive';
export const REACTIVE_INVESTIGATION = 'Reactive Investigation';
export const REQUIRED = 'Required';
export const SAVING = 'Saving';
export const PLANS = 'Manage Plans';
export const PLANNING = 'Planning';
export const MONITOR = 'Monitor';
export const ADMIN = 'Admin';
export const DRAFTS_PARENTHESIS = '(drafts)';
export const CONFIRM = 'Confirm';
export const CANCEL = 'Cancel';
export const JURISDICTION = 'Jurisdiction';
export const ADMIN_LEVEL = 'Admin Level';
export const DISTRICT = 'District';
export const CANTON = 'Canton';
export const FI_REASON = 'FI Reason';
export const FI_STATUS = 'FI Status';
export const MEASURE = 'Measure';
export const MARK_AS_COMPLETE = 'Mark as complete';
export const STRUCTURES = 'structure(s)';
export const PERSONS = 'person(s)';

export const FOCUS_AREA_HEADER = 'Focus Area';
export const SPRAY_AREA_HEADER = 'Spray Area';
export const INTERVENTION_TYPE_LABEL = 'Intervention Type';
export const FOCUS_INVESTIGATION_STATUS_LABEL = 'Focus Investigation Status';
export const FOCUS_INVESTIGATION_STATUS_REASON = 'Focus Investigation Reason';
export const CASE_NUMBER = 'Case Number';
export const LAST_MODIFIED = 'Last Modified';
export const TITLE = 'Title';
export const PLAN_TITLE_LABEL = 'Plan Title';
export const PLAN_START_DATE_LABEL = 'Plan Start Date';
export const PLAN_END_DATE_LABEL = 'Plan End Date';
export const ACTIVITIES_LABEL = 'Activities';
export const DESCRIPTION_LABEL = 'Description';
export const QUANTITY_LABEL = 'Quantity';
export const PRIORITY_LABEL = 'Priority';
export const ADD_ACTIVITY = 'Add Activity';
export const REASON_HEADER = 'Reason';
export const STATUS_HEADER = 'Status';
export const CASE_NOTIF_DATE_HEADER = 'Case Notif. Date';
export const CASE_CLASSIFICATION_HEADER = 'Case Class.';
export const CASE_CLASSIFICATION_LABEL = 'Case Classification';
export const DEFINITIONS = 'Definitions';
export const TYPE_LABEL = 'Type';
export const LOCATIONS = 'Locations';

export const PROVINCE = 'Province';
export const HOME = 'Home';
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export const CURRENT_FOCUS_INVESTIGATION = 'Current Focus Investigations';
export const COMPLETE_FOCUS_INVESTIGATION = 'Complete Focus Investigations';
export type COMPLETE_FOCUS_INVESTIGATION = typeof COMPLETE_FOCUS_INVESTIGATION;
export const INVESTIGATION = 'Investigation';
export const INTERVENTION = 'Intervention';
export const FOCUS_AREA_INFO = 'Focus Area Information';
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export const ASSIGN_PRACTITIONERS_TO_ORG = 'Assign Practitioners to %s';
export const DATE_IS_REQUIRED = 'Date is Required';
export const NAME_IS_REQUIRED = 'Name is Required';
export const NUMERATOR_OF_DENOMINATOR_UNITS = '%s of %s %s';
export const MARK_PLAN_AS_COMPLETE = 'Mark %s as complete';
export const FIS_IN_JURISDICTION = 'Focus Investigations in %s';
export const FI_IN_JURISDICTION = 'Current Focus Investigations in %s';
export const PROGRESS = 'Progress';
export const ASSIGN_JURISDICTIONS = 'Assign Jurisdictions';
export const SELECT_JURISDICTIONS = 'Select Jurisdictions';
export const ADD_CODED_ACTIVITY = 'Add %s Activity';
export const ORGANIZATION_DETAILS = 'Team Details';
export const TEAM_MEMBERS = 'Team Members';
export const ONE_DAY_TO_GO = '1 day to go';
export const ZERO_DAYS_TO_GO = '0 days to go';
export const PRACTITIONERS_ASSIGNED_TO_ORG = '%s Practitioners successfully assigned to %s';
export const PRACTITIONER_REMOVED_FROM_ORG = '%s-%s successfully removed from %s';
export const ORGANIZATION_EDITED_SUCCESSFULLY = `Team edited successfully.`;
export const ORGANIZATION_CREATED_SUCCESSFULLY = `Team created successfully`;
export const DRAFT_PLAN_TITLE = '%s (draft)';
export const EDIT_PRACTITIONER = 'Edit Practitioner';
export const NEW_PRACTITIONER = 'New Practitioner';
export const SAVE_PRACTITIONER = 'Save Practitioner';

export const ROUTINE_TITLE = 'Routine';
export const ROUTINE_INVESTIGATION_TITLE = 'Routine Investigation';
export const BCC_ACTIVITY = 'Behaviour Change Communication';
export const BCC_ACTIVITY_DESCRIPTION = 'Conduct BCC activity';
export const BCC_GOAL_DESCRIPTION = 'Complete at least 1 BCC activity for the operational area';
export const BCC_GOAL_MEASURE = 'Number of BCC Activities Completed';
export const IRS_ACTIVITY = 'Spray Structures';
export const IRS_ACTIVITY_DESCRIPTION =
  'Visit each structure in the operational area and attempt to spray';
export const IRS_GOAL_DESCRIPTION = 'Spray structures in the operational area';
export const IRS_GOAL_MEASURE = 'Percent of structures sprayed';
export const BEDNET_ACTIVITY = 'Bednet Distribution';
export const BEDNET_ACTIVITY_DESCRIPTION =
  'Visit 100% of residential structures in the operational area and provide nets';
export const BEDNET_GOAL_MEASURE = 'Percent of residential structures received nets';
export const BLOOD_SCREENING_ACTIVITY = 'Blood screening';
export const BLOOD_SCREENING_ACTIVITY_DESCRIPTION =
  'Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person';
export const BLOOD_SCREENING_GOAL_MEASURE = 'Number of registered people tested';
export const CASE_CONFIRMATION_ACTIVITY = 'Case Confirmation';
export const CASE_CONFIRMATION_ACTIVITY_DESCRIPTION = 'Confirm the index case';
export const CASE_CONFIRMATION_ACTIVITY_GOAL_MEASURE = 'Number of cases confirmed';
export const RACD_ACTIVITY = 'Family Registration';
export const RACD_ACTIVITY_DESCRIPTION =
  'Register all families & family members in all residential structures enumerated (100%) within the operational area';
export const RACD_ACTIVITY_GOAL_MEASURE =
  'Percent of residential structures with full family registration';
export const LARVAL_DIPPING_ACTIVITY = 'Larval Dipping';
export const LARVAL_DIPPING_ACTIVITY_DESCRIPTION =
  'Perform a minimum of three larval dipping activities in the operational area';
export const LARVAL_DIPPING_GOAL_MEASURE = 'Number of larval dipping activities completed';
export const MOSQUITO_COLLECTION_ACTIVITY = 'Mosquito Collection';
export const MOSQUITO_COLLECTION_ACTIVITY_DESCRIPTION =
  'Set a minimum of three mosquito collection traps and complete the mosquito collection process';
export const MOSQUITO_COLLECTION_GOAL_MEASURE =
  'Number of mosquito collection activities completed';

// TODO ? - do the below 2 belong here or in a settings file
export const ORGANIZATION_LABEL = TEAM;
export const ORGANIZATIONS_LABEL = TEAMS;
