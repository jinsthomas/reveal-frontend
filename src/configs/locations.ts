/** Interface that describes location items */
export interface LocationItem {
  identifier: string /** Should match the name of the column in data */;
  level: number /** The HDX-compliant level of the location in the hierarchy */;
  name: string /** The name of the location */;
}

/** Location item hierarchy
 * This is a list of locations.  The "level" field will be used to sort the
 * locations hierarchically, from lowest to highest.
 */
export const locationHierarchy: LocationItem[] = [
  {
    identifier: 'province',
    level: 1,
    name: 'Province',
  },
  {
    identifier: 'district',
    level: 2,
    name: 'District',
  },
  {
    identifier: 'canton',
    level: 3,
    name: 'Canton',
  },
  {
    identifier: 'village',
    level: 4,
    name: 'Village',
  },
];
