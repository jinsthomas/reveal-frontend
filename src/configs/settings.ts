/** This is the main configuration module */
import { Providers } from '@onaio/gatekeeper';
import {
  DOMAIN_NAME,
  ONADATA_ACCESS_TOKEN_URL,
  ONADATA_AUTHORIZATION_URL,
  ONADATA_CLIENT_ID,
  ONADATA_OAUTH_STATE,
  ONADATA_USER_URL,
} from './env';

/** Interfaces and Types */

/** Interface for a Focus Investigation Classification */
export interface Classification {
  code: string;
  name: string;
  description: string;
}

/** Interface that describes location items */
export interface LocationItem {
  identifier: string /** Should match the name of the column in data */;
  level: number /** The HDX-compliant level of the location in the hierarchy */;
  name: string /** The name of the location */;
}

/** Authentication Configs */
export const providers: Providers = {
  Ona: {
    accessTokenUri: ONADATA_ACCESS_TOKEN_URL,
    authorizationUri: ONADATA_AUTHORIZATION_URL,
    clientId: ONADATA_CLIENT_ID,
    redirectUri: `${DOMAIN_NAME}/oauth/callback/Ona/`,
    scopes: ['read', 'write'],
    state: ONADATA_OAUTH_STATE,
    userUri: ONADATA_USER_URL,
  },
};

/** Location configs */

/** Location item hierarchy
 * This is a list of locations.  The "level" field will be used to sort the
 * locations hierarchically, from lowest to highest.
 */
export const locationHierarchy: LocationItem[] = [
  {
    identifier: 'province',
    level: 1,
    name: 'Province',
  },
  {
    identifier: 'district',
    level: 2,
    name: 'District',
  },
  {
    identifier: 'canton',
    level: 3,
    name: 'Canton',
  },
  {
    identifier: 'village',
    level: 4,
    name: 'Village',
  },
];

/** Focus investigation configs */

/** Focus Investigation case classifications */
export const FIClassifications: Classification[] = [
  {
    code: 'A1',
    description: 'Indigenous case recorded within the last year.',
    name: 'Active',
  },
  {
    code: 'A2',
    description: 'No indigenous case during the last year, but withing the last 3 years.',
    name: 'Residual Non-Active',
  },
  {
    code: 'B1',
    description: 'Receptive area but no indigenous cases within the last 3 years.',
    name: 'Cleared Receptive',
  },
  {
    code: 'B2',
    description: 'Non-receptive area.',
    name: 'Cleared Non-Receptive',
  },
];

/** Indicators configs */

// thresholds
export const GREEN_THRESHOLD = 0.9;
export type GREEN_THRESHOLD = typeof GREEN_THRESHOLD;
export const YELLOW_THRESHOLD = 0.2;
export type YELLOW_THRESHOLD = typeof YELLOW_THRESHOLD;
export const ORANGE_THRESHOLD = 0.8;
export type ORANGE_THRESHOLD = typeof ORANGE_THRESHOLD;

// 1-3-7 thresholds
export const ONE = 0;
export const ZERO = 0;
