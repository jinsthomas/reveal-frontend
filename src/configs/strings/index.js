import * as en from './en.json';
import * as sw from './sw.json';

export default {
  en,
  sw,
};
