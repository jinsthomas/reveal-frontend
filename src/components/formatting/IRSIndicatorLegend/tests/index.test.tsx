import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import IRSIndicatorLegend from '..';
import { indicatorThresholdsNA } from '../../../../configs/settings';

describe('/components/formatting/IRSIndicatorLegend', () => {
  it('renders correctly', () => {
    const wrapper = mount(<IRSIndicatorLegend />);
    expect(wrapper.props()).toEqual({ indicatorThresholds: indicatorThresholdsNA });
    expect(toJson(wrapper.find('h5'))).toMatchSnapshot('header');
    expect(toJson(wrapper.find('Table'))).toMatchSnapshot('indicator table');
    wrapper.unmount();
  });
});
