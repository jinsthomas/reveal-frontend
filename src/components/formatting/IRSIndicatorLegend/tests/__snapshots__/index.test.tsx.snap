// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/formatting/IRSIndicatorLegend renders correctly: header 1`] = `
<h5
  className="mt-4 mb-3"
>
  IRS Conditional Formatting
</h5>
`;

exports[`/components/formatting/IRSIndicatorLegend renders correctly: indicator table 1`] = `
<Table
  className="text-center"
  responsiveTag="div"
  tag="table"
>
  <table
    className="text-center table"
  >
    <tbody>
      <tr
        key="0"
        style={
          Object {
            "background": "#dddddd",
          }
        }
      >
        <td>
          Grey
        </td>
        <td>
          &gt;=
        </td>
        <td>
          20%
        </td>
      </tr>
      <tr
        key="1"
        style={
          Object {
            "background": "#FF4136",
          }
        }
      >
        <td>
          Red
        </td>
        <td>
          &gt;=
        </td>
        <td>
          75%
        </td>
      </tr>
      <tr
        key="2"
        style={
          Object {
            "background": "#FFDC00",
          }
        }
      >
        <td>
          Yellow
        </td>
        <td>
          &gt;=
        </td>
        <td>
          90%
        </td>
      </tr>
      <tr
        key="3"
        style={
          Object {
            "background": "#2ECC40",
          }
        }
      >
        <td>
          Green
        </td>
        <td>
          &gt;=
        </td>
        <td>
          100%
        </td>
      </tr>
    </tbody>
  </table>
</Table>
`;
